### ISSUE TITLE
Getting false (?) errors "log_error:307"

### ISSUE THREAD

**Issue by hettervik** [domeger/SplunkTAforPuppetEnterprise#10]
Hi,

I've ran the Puppet TA for a while now, and after fixing some timezone issues, it have been working just fine. There is one problem though, I'm getting _a lot_ of error messages in the internal logs from the TA. These error messages seems to be false positives (should be INFO). I'm suspecting that the TA logs an error message in the internal log for every Puppet event the TA collects. The errors looks like the following, with a different stuff in the square brackets.

`ERROR pid=12345 tid=MainThread file=base_modinput.py:log_error:307 | [...]`

This same issue was also mentioned in my previous thread (https://github.com/mrzarquon/SplunkTAforPuppetEnterprise/issues/6), although the main question was another, so the issue was closed.

Is it something wrong with my setup, og is the TA indeed incorrectly logging INFO logs as ERROR logs?

**Comment by mrzarquon** [comment]
Hey @hettervik - sorry for the long delay on this. We've found that pulling from PuppetDB ends up not scaling as we'd hope. We've added support to Puppet via a module that allows us to publish information via the HEC. So far we have moved most of the functionality over to this method, submitting facts and reports. The Splunk App is here: https://splunkbase.splunk.com/app/4413/ which contains the dashboards and sourcetypes, while the module for sending the data is here (with a pending 0.5.0 update this week): https://forge.puppet.com/puppetlabs/splunk_hec

**Comment by hettervik** [comment]
Hi! Thanks a lot for the update! We've finally got to install and set up the new app that you've made, along with the HEC, and it seems to be working smoothly. I haven't tested all the features yet. As far as I can understand not all of the logs are obtainable through the HEC, some has to be collected with API calls to the PuppetDB as well? We have only tested receiving logs with the HEC, not the API calls.

Noticed that the documentation for the Puppet module wrongly label the HEC as the HTTP Endpoint Collector. It's supposed to be HTTP Event Collector. Heh.

After we've tested the log collection and dashboards some more, I'll surely get back to you if we have any suggestions for improvements. Thanks!
